packages/eas-cli/src/commands/branch/delete.ts

```diff
@@ -0,0 +1,180 @@
+import { getConfig } from '@expo/config';
+import { Command, flags } from '@oclif/command';
+import chalk from 'chalk';
+import gql from 'graphql-tag';
+
+import { graphqlClient, withErrorHandlingAsync } from '../../graphql/client';
+import { DeleteUpdateBranchResult, Update } from '../../graphql/generated';
+import Log from '../../log';
+import { ensureProjectExistsAsync } from '../../project/ensureProjectExists';
+import { findProjectRootAsync, getProjectAccountName } from '../../project/projectUtils';
+import { promptAsync, toggleConfirmAsync } from '../../prompts';
+import { ensureLoggedInAsync } from '../../user/actions';
+
+async function getBranchInfoAsync({
+  appId,
+  name,
+}: {
+  appId: string;
+  name: string;
+}): Promise<{ branchId: string }> {
+  const data = await withErrorHandlingAsync(
+    graphqlClient
+      .mutation<
+        {
+          app: {
+            byId: {
+              updateBranchByName: {
+                id: string;
+                name: string;
+                updates: Pick<Update, 'id'>[];
+              };
+            };
+          };
+        },
+        {
+          appId: string;
+          name: string;
+        }
+      >(
+        gql`
+          query ViewBranch($appId: String!, $name: String!) {
+            app {
+              byId(appId: $appId) {
+                id
+                updateBranchByName(name: $name) {
+                  id
+                  name
+                }
+              }
+            }
+          }
+        `,
+        {
+          appId,
+          name,
+        }
+      )
+      .toPromise()
+  );
+
+  const {
+    app: {
+      byId: {
+        updateBranchByName: { id: branchId },
+      },
+    },
+  } = data;
+  return { branchId };
```

Comment:
```suggestion
  return { branchId: data.app.byId.updateBranchByName.id };
```
Seems less convoluted than destructuring in this case.